### PR TITLE
Talent UI Updates

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

Talent UI Updates
- Move Account Info location in Captive Portal
- Include Translations in Captive Portal
- Show Renewal Date in Captive Portal
- Show Renewal Date in User Training Admin Page

Bug Fix
- Fix issue with becoming user whose account had been deleted in talent but not cleared out locally

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected

### CODE REVIEW

**Comment by dtgreiner** [comment]
User Training Admin Page UI
Screenshot 2024-11-22 at 11 52 02 AM

Will add Talent Captive Portal screenshot after API count is reset

File: app/models/concerns/user_concern.rb [hunk 1]
```diff
@@ -226,6 +226,22 @@ def training_status(course)
...
+      login = Talentlms::Login.find_by(user: self, config: course.config)
+      return nil unless login
+
+      completion_date = Talentlms::CompletedTraining.find_by(course: course, login: login)&.completion_date
+      return nil unless completion_date
+
+      if Talentlms::Facade.expiration_duration_period == :days
+        completion_date + course.months_to_expiration
+      else
+        completion_date << course.months_to_expiration
```
Comment (eanders) [inline app/models/concerns/user_concern.rb]: Does the shovel operator have some different behavior for dates?  Is this different from:
```
if Talentlms::Facade.expiration_duration_period == :days
         completion_date + course.months_to_expiration.days
       else
         completion_date + course.months_to_expiration.months
```

File: app/views/admin/user_trainings/edit.haml [hunk 2]
```diff
@@ -21,9 +21,10 @@
...
         %th Progress
+        %th Renewal Required After
     %tbody
       - @user.required_training_courses.each do |course|
         %tr
           %td= course.name_with_subdomain
           %td= @user.training_status(course)
-    
\ No newline at end of file
+          %td= @user.training_renewal_date(course)
```
Comment (eanders) [inline app/views/admin/user_trainings/edit.haml:30]: Maybe this should say `not required` or `never` if the date is blank?

File: app/models/concerns/user_concern.rb (hunk 1 above)
Comment (dtgreiner) [reply app/models/concerns/user_concern.rb]: The shovel operator adds months to a date, but I'm going to change it to your recommendation. For whatever reason this way of doing it wasn't coming to me when building this out.

File: app/views/admin/user_trainings/edit.haml (hunk 2 above)
Comment (dtgreiner) [reply app/views/admin/user_trainings/edit.haml:30]: I'll update the user concern method to return 'Never' when the `months_to_expiration` field is nil. This should allow it to still be blank when there is an expiration but the training hasn't been completed.

**Comment by dtgreiner** [comment]
Updated Screenshots
Screenshot 2024-11-22 at 1 05 08 PM
Screenshot 2024-11-22 at 1 11 48 PM

**Review by eanders — APPROVED** [review]
Thanks @dtgreiner !

**Comment by dtgreiner** [comment]
@eanders This is ready for another look. Thanks!

**Pull request merged**